go/common/witness/proof.go

```diff
@@ -0,0 +1,66 @@
+package witness
+
+import (
+	"github.com/Fantom-foundation/Carmen/go/common"
+	"github.com/Fantom-foundation/Carmen/go/common/tribool"
+)
+
+// Proof is an interface for witness proofs. It allows for adding proofs and checking if the proof is valid.
+type Proof interface {
+
+	// IsValid checks that this proof is self-consistent. If the result is true, the proof can be used
+	// for extracting verified information. If false, the proof is corrupted and should be discarded.
+	IsValid() bool
+
+	// Add adds a proof node to the witness proof.
+	Add(Proof)
+}
+
+// ProofState allows for getting properties of accounts and account storage slots.
+// It contains methods to determine values of properties based on addresses and keys,
+// and returns the values and information if the proof was sufficient to determine the value.
+type ProofState interface {
```

Comment:
Why the differentiation between a `Proof` and a `ProofState`. Could this not be the same?

Reply:
I was at first sticked to the idea that you create a proof first --> you get `Proof`, and "later" you query the proof (for a root hash), which gives you the `ProofState` -> you query this. 

I think it sounds plausible to merge these two together, perhaps I will try to implement and see how would this look like